litmus-portal/graphql-server/pkg/graphql/mutations/mutation.go

```diff
@@ -54,6 +61,7 @@ func ClusterRegister(input model.ClusterInput) (*model.ClusterRegResponse, error
 		ClusterID:   newCluster.ClusterID,
 		Token:       token,
 		ClusterName: newCluster.ClusterName,
+		Endpoint:    endPoint,
```

Comment:
There is no need to persist it in the DB as the endpoint will be the same everytime?

Reply:
No endpoint will not same every time. It depends which node the pod had been schedule.

Reply:
Then the whole external cluster thing will fail, if the ip keeps changing how will the subscriber connect?

Reply:
Also, it required to save in db because after logout frontend will lose the state

Reply:
Yes this is a existing problem. Pod can be schedule to another node.if the node has some problem

Reply:
Backend can always send the same endpoint, so state management is not required. But this is a bigger problem if the pod is rescheduled all external clusters are disconnected? We can't use this node ip approach then

Reply:
This can be solved by ingress , for now it's okay. Where will frontend stores the endpoint if we logout?

Reply:
It is required for the getcluster query, when user wants to see the pending cluster endpoint

Reply:
If the endpoint is constant we can just retrieve that endpoint and always provide frontend with that, so no need to store. We should use ingress rather than nodeIP then. This will cause problems.

Reply:
And also we should be sending the updated endpoint and not the old nodeIP which could've changed. Storing in db will give old endpoint which will be useless.

Reply:
Ingress is one approach, other approach is load balancer. In ingress, user will need a domain to setup. It's not easy to setup.

Reply:
Even if we don't want to setup ingress or loadbalancer right now, storing nodeIP in db will not help as I already stated above.